### DISCUSSION TITLE
Json token position library?

### DISCUSSION THREAD

**Discussion by tonka3000** [microsoft/vscode-discussions#7] [Extension Development QnA] [answered]
Hi,

I'm searching for a API/library to get the position of a specific token in a json file. My goal would be to jump to it via command, so I need a "token to position" API.

Anybody know if there is a builtin feature in VSCode or any good node library.

Thanks in advance

**Comment by isidorn** [comment]
I do not think we expose the low level tokens in API. So I believe this is not possible.
Can you explain your use case in a bit more detail. Just for clarity. Thank you

**Reply by tonka3000** [reply]
Hi @isidorn ,

I have a treeview where I represent parts of a json file content to make it easier for users to navigate in that file (right now, it is `package.json`, but I would love to use it for other json files as well).

My main problem is that I need to have a way to map the token in the file to a position in the textfile.

example:
```json
{
    "key" : {
        "subkey" : "value"
   }
}
```

The API should be able to to the following

```ts
// "jsonfileAPI" which already read in a json file
const subkey = jsonfileAPI.get("subkey");
console.log(subkey.value ); // print 'value'
console.log(subkey.startPosition); // print ''20" when it count all characters or "[3,4]" when it would be the position via column,line
```

That way I would be able to jumo to the position.

This is what outline view is doing.

**Comment by digitarald** [comment, marked as answer]
You'll need to parse an AST from the JSON source, like https://github.com/vtrushin/json-to-ast , which will have position information for each node.

**Reply by tonka3000** [reply]
Thanks @digitarald , this is exactly what I was looking for.

**Comment by robconery** [comment]
Glad you found your answer! Closing...
